draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1551,76 +1555,92 @@ user agent might wish to block sending cookies during "third-party" requests
 from setting cookies (see {{third-party-cookies}}).
 
 If the user agent does attach a Cookie header field to an HTTP request, the
-user agent MUST send the cookie-string (defined below) as the value of the
-header field.
+user agent MUST send the cookie-string as the value of the
+header field. The cookie-string is computed as defined in {{retrieval-algorithm}},
+whereby the retrieval-uri is defined as the request-uri and the same-site status
+is computed for the HTTP request as defined in {{same-site-requests}}.
 
-The user agent MUST use an algorithm equivalent to the following algorithm to
-compute the cookie-string from a cookie store and a request-uri:
+### Non-HTTP APIs {#non-http}
+
+The user agent MAY implement "non-HTTP" APIs that can be used to access
+stored cookies.
 
-1.  Let cookie-list be the set of cookies from the cookie store that meets all
-    of the following requirements:
+A user agent MAY return an empty cookie-string in certain contexts, such as when
+retrieval occurs within a third-party context (see {{third-party-cookies}}).
 
-    *   Either:
+If a user agent does return cookies for a given call to a "non-HTTP" API with an
+associated Document, then the user agent MUST compute the cookie-string following
+the algorithm defined in {{retrieval-algorithm}, whereby the retrieval-uri is the
+associated Document's URL. The retrieval is same-site if the Document's "site for
+cookies" is same-site with the top-level origin as defined in {{document-requests}}.
 
-        *   The cookie's host-only-flag is true and the canonicalized
-            request-host is identical to the cookie's domain.
+### Retrieval Algorithm {#retrieval-algorithm}
 
-        Or:
+Given a cookie store, a retrieval-uri, and the same-site status of the
+retrieval-uri, the following algorithm returns a cookie-string.
```

Comment:
> Also, I'm not sure "same-site status of the retrieval-uri" sounds quite right, (I actually disagree with Mike), I think "same-site status of the retrieval" is better. (Because the same-site status is a function of more than just the uri.) We're already using the term retrieval below anyway. And we can define retrieval as any event that requires generating a cookie-string, or something like that.

I think you're right. I've reverted back to "same-site status of the retrieval" and defined a "retrieval" at the start of the section.

Reply:
> IIUC the way "same-site status" is computed is not defined in 6265bis for non-HTTP retrievals, but is an input to this algorithm from elsewhere. Should that be called out in the section above?

In the end I decided to keep the non-HTTP API subsection which includes a description of how to compute the same-site status of the retrieval:

> retrieval is from a "non-HTTP" API. The retrieval-uri is the associated
> Document's cookie URL {{COOKIE-URL}}, and the retrieval is same-site if the
> Document's "site for cookies" is same-site with the top-level origin as defined
> in {{document-requests}}.